test/e2e/edit-app-settings.spec.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const path = require('path');
+const { expect } = require('chai');
+const request = require('request-promise-native');
+
+const { getProjectUrl } = require('./cht-docker-utils');
+const {
+  DEFAULT_PROJECT_NAME,
+  cleanupProject,
+  getProjectDirectory,
+  initProject,
+  runChtConf,
+} = require('./cht-conf-utils');
+
+describe('edit-app-settings', () => {
+  const projectName = DEFAULT_PROJECT_NAME;
+  const projectDirectory = getProjectDirectory(projectName);
+
+  before(async () => {
+    await initProject(projectName);
+  });
+
+  after(async () => {
+    await cleanupProject(projectName);
+  });
+
+  it('disables a language, recompile, and push app settings', async () => {
+    const url = await getProjectUrl(projectName);
+    const initialSettings = await request.get({ url: `${url}/api/v1/settings`, json: true });
+
+    // TODO: remove next line when we upgrade eslint and its `parserOptions.ecmaVersion` setting to parse syntax supported by node 18+
+    // eslint-disable-next-line no-undef
+    const baseSettings = structuredClone(initialSettings);
```

Comment:
What is the need of doing a structuredClone if you are never referencing `initialSettings` again?

Reply:
In the first iterations I was reusing `initialSettings` and that led me to clone it with `structuredClone` and I left it there after changing the code. It's not useful anymore so I'll get rid of it, nice catch

Reply:
I am excited to see `structuredClone` actually getting some use! Definitely been needing that function (but I guess not in this case...)!

![shoul](https://i.kym-cdn.com/entries/icons/original/000/040/653/goldblum-quote.jpeg)